### PR TITLE
Use C++14 aliases for some type traits

### PR DESCRIPTION
**Pull request by rschu1ze**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionMLMethod.cpp [hunk 1]
```diff
@@ -83,11 +83,11 @@ namespace
...
-        if (std::is_same<Method, FuncLinearRegression>::value)
+        if (std::is_same_v<Method, FuncLinearRegression>)
         {
             gradient_computer = std::make_unique<LinearRegression>();
         }
-        else if (std::is_same<Method, FuncLogisticRegression>::value)
+        else if (std::is_same_v<Method, FuncLogisticRegression>)
         {
             gradient_computer = std::make_unique<LogisticRegression>();
         }
```
Comment (antonio2368) [inline src/AggregateFunctions/AggregateFunctionMLMethod.cpp:93]: I think this should be constexpr check, and in else `static_assert(false, "Such gradient computer is not implemented yet")`.

Can you fix it as part of this PR?

Comment (rschu1ze) [reply src/AggregateFunctions/AggregateFunctionMLMethod.cpp:93]: Good catch, tnx. Will fix it.

Comment (rschu1ze) [reply src/AggregateFunctions/AggregateFunctionMLMethod.cpp:93]: Uh ... a static_assert in the else branch would need an ugly workaround (--> https://stackoverflow.com/questions/53945490/how-to-assert-that-a-constexpr-if-else-clause-never-happen). I changed to 2x constexpr checks but would rather keep the current throw in the else. As the else cannot be reached, we wouldn't improve performance anyways.

Comment (rschu1ze) [reply src/AggregateFunctions/AggregateFunctionMLMethod.cpp]: Okay, I found a C++20 solution for the static_assert which looks acceptable imho.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**